### ISSUE TITLE
AzureMapHtmlMarker options.popupContent doesn't seem allow for click events anywhere in the content

### ISSUE THREAD

**Issue by ronvecchi** [WiredSolutions/react-azure-maps#57]
I tried adding a div with an onClick handler to the popContent and it seems it's never invoked.  Almost like the event is never getting registered.

To verify and eliminate any thing that I might have been doing.  I took the **PopupExample.tsx** from the playground and added an onClick handler to the div:

```
  const onPopupCLick = (): void => {	
	console.log('I got clicked');
  };

  const htmlMarkerOptions = {
    popup: useCreatePopup({
      options: {},
      popupContent: <div style={wrapperStyles.popupStyles} **onClick={() => onPopupCLick()}** >Html marker popup with some random value click Change Number to see actual value <br /> {someHtmlMarkerPopupState}</div>
    })
  }
```

It never get's invoked when clicking the div.

Side note:  Also seems to be the same issue with the AzureMapPopup

**Comment by ronvecchi** [comment]
Seems to be related to the react-dom/server call to renderToStaticMarkup() in useCreatePopup().

**Comment by ronvecchi** [comment]
Ok so looked at closed issue 54 which I can see that it's not possible...
https://github.com/WiredSolutions/react-azure-maps/issues/54

Hack to get around this was to supply a div to the content with an Id and then use ReactDOM.render within an effect to update the content.

Something like:

```
                    <AzureMapPopup
                        key={rendId}
                        isVisible={true}
                        popupContent={(<div id="myPopupDiv"></div>)}
                        events={[{
                            eventName: 'close',
                            callback: onClosePopup
                        }]}
                        options={{
                            closeButton: true,
                            position: [poiResult?.position?.lon ?? 0, poiResult?.position?.lat ?? 0]
                        }}
                    />
```

And then for the effect:

```
    useEffect(() => {
        const element = document.getElementById('myPopupDiv');

        if (element) {
            const poiResult = (nearbyMarkerInfo && nearbyMarkerInfo.results && nearbyMarkerInfo.results.length > 0 && poiResultId)
                ? nearbyMarkerInfo.results.find(r => r.id === poiResultId)
                : undefined;

            ReactDOM.render(<TripPoiPopupComponent
                item={poiResult}
            />, element)
        }

    }, [nearbyMarkerInfo, poiResultId]);

```

**Comment by msasinowski** [comment]
This is a good approach @ronvecchi and we probably look into this, but I see at least one concern. It will probably affect the performance of this component because we will need to render the whole tree under the `TripPoiPopupComponent` on every `[nearbyMarkerInfo, poiResultId]` change. Which is probably fine but not in every use case. I will not close this issue, and try this approach on some more complex examples.

**Comment by zech** [comment]
I've used a similar approach. But instead of rendering a whole new React "instance" I've used ReactDOM.createPortal().
This way all events bubble as usual throug the tree.

```jsx
  <>
    <AzureMap {...props}>
      <AzureMapPopup
        isVisible={isPopupVisible}
        options={popupOptions}
        popupContent={<div id="myPopup" />}
      />
    </AzureMaps>
    { isPopupVisible && <PopupPortalComponent {...popupProps} /> }
  </>
```

```jsx
const PopupPortalComponent = (props) => {
    const popup = document.getElementById("myPopup");
    if (popup == null) {
        return null;
    }

    return ReactDOM.createPortal(<PopupContainer {...props} />, popup);
};
```

But in my case I had issues with rendering the popup the first time which I could not track down yet so I helped my self out with a hacky setTimeout() triggering a rerender by flipping a state property once.

Aside from that everything seems to work so far.

**Comment by psrednicki** [comment]
@zech Can you add code where you add this hacky setTimeout? Maybe map isn't ready yet. You can get `mapRef` and `isMapReady` form `AzureMapsContext`, maybe this will resolve your issue.

**Comment by zech** [comment]
That's another issue I'm struggling with. Due to legacy code I cannot use TS so I adapted the example code for getting the AzureMapsContext.

Currently my hack looks like this:

```jsx
    useEffect(
        () => {
            // just trigger a rerender because popup does not show
            // on first glance
            setTimeout(setDoReRender, 0, !doReRender);
        },
        [isPopupVisible, popupProps]
    );
```

It has to be done only once.

To get the mapRef I'm trying this:

```js
    const { mapRef, isMapReady } = useContext(AzureMapsContext);
    if (isMapReady) {
        debugger;
    }
```

But the debugger never triggers. But I don't want to hijack this topic on this matter.

**Comment by psrednicki** [comment]
Thats weird. Are you sure you wrap your component with `<AzureMapsProvider>`?
You can use it in JS as well.
Maybe you can create another issue? We will have more context for other users and I will try to help you.
